app/src/examples/PendulumExample.tsx

```diff
@@ -0,0 +1,233 @@
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withSpring,
+} from 'react-native-reanimated';
+import { StyleSheet, TextInput, Text } from 'react-native';
+import { View, Button } from 'react-native';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import {
+  Gesture,
+  GestureDetector,
+  GestureHandlerRootView,
+  GestureStateManager,
+  GestureTouchEvent,
+} from 'react-native-gesture-handler';
+
+type FieldDefinition = [
+  string,
+  string,
+  Dispatch<SetStateAction<string>>,
+  Dispatch<SetStateAction<number>>
+];
+
+function InputField({
+  fieldName,
+  textValue,
+  setTextValue,
+  setValue,
+}: {
+  fieldName: string;
+  textValue: string;
+  setTextValue: Dispatch<SetStateAction<string>>;
+  setValue: Dispatch<SetStateAction<number>>;
+}) {
+  return (
+    <View style={styles.row} key={fieldName}>
+      <Text
+        style={[
+          styles.inputDescription,
+          {
+            textDecorationLine: fieldName === 'Duration' ? 'underline' : 'none',
+            fontWeight: fieldName === 'Duration' ? 'bold' : undefined,
+          },
+        ]}>
+        {fieldName}
+      </Text>
+      <TextInput
+        key={fieldName}
+        value={textValue}
+        onChangeText={(s) => {
+          const parsedInput = Number.parseFloat(s);
+          if (parsedInput) {
+            setValue(parsedInput);
+          }
+          setTextValue(s);
+        }}
```

Comment:
Can we move it outside and memoize it?

Reply:
I'm afraid we can't, since I've moved InputField out of function SpringExample to reduce number re-renders (which make textInput loose its focus)